### PR TITLE
Secret variables

### PR DESCRIPTION
**Pull request by brenthdaniel**
Tests and code for #9786

### CODE REVIEW

**Comment by brenthdaniel** [comment]
#build

**Comment by brenthdaniel** [comment]
Missed a unit test change, 
#build again

**Comment by brenthdaniel** [comment]
infrastructure problems... 
#build again

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops [hunk 1]
```diff
@@ -476,5 +476,10 @@ error.root.must.be.server=CWWKG0105E: The {0} configuration document must begin
 error.root.must.be.server.explanation=The configuration must be contained in either a server element or a client element. 
 error.root.must.be.server.useraction=Add a server or client element as the root of the configuration file. 
 
+error.bad.variable.file=CWWKG0106E: The file {0} can not be read.
```
Comment (dmuelle) [inline dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops:479]: **change**
`The file {0} can not be read. `
**to**
`The {0} file can not be read.`

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops [hunk 2]
```diff
@@ -476,5 +476,10 @@ error.root.must.be.server=CWWKG0105E: The {0} configuration document must begin
 error.root.must.be.server.explanation=The configuration must be contained in either a server element or a client element. 
 error.root.must.be.server.useraction=Add a server or client element as the root of the configuration file. 
 
+error.bad.variable.file=CWWKG0106E: The file {0} can not be read. 
+error.bad.variable.file.explanation=The file can not be loaded as a variable because it can not be read. 
+error.bad.variable.file.useraction=Verify that the file contains text content and check operating system permissions on the file.
```
Comment (dmuelle) [inline dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops:481]: **change**
Verify that the file contains text content and check operating system permissions on the file. 
**to**
Verify that the file contains text content and check the operating system permissions for the file.

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops [hunk 3]
```diff
@@ -476,5 +476,10 @@ error.root.must.be.server=CWWKG0105E: The {0} configuration document must begin
 error.root.must.be.server.explanation=The configuration must be contained in either a server element or a client element. 
 error.root.must.be.server.useraction=Add a server or client element as the root of the configuration file. 
 
+error.bad.variable.file=CWWKG0106E: The file {0} can not be read. 
+error.bad.variable.file.explanation=The file can not be loaded as a variable because it can not be read.
```
Comment (dmuelle) [inline dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops:480]: is it possible to say why the file can't be read or what cannot read it?

**Review by dmuelle — CHANGES_REQUESTED** [review]
a few minor edits.  For the explanation, if it's possible to suggest common causes, or to say what agent cannot read the file we should include that info. Let me know if any questions, thanks

**Comment by andrea-pichler** [comment]
Completed the requested L2 message review

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops (hunk 3 above)
Comment (brenthdaniel) [reply dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops:480]: Not really.. We just know that something went wrong. The most likely causes are in the user action.

**Comment by brenthdaniel** [comment]
#build

**Review by dmuelle — APPROVED** [review]
LGTM

**Review by hlhoots — APPROVED** [review]
Looks good based on the team review that we did on scrum today.

**Comment by brenthdaniel** [comment]
#build

**Comment by brenthdaniel** [comment]
#build

**Comment by brenthdaniel** [comment]
#build

**Comment by brenthdaniel** [comment]
#build

**Comment by brenthdaniel** [comment]
#run-libby-bot

**Pull request merged**